static_src/components/route_form.jsx

```diff
@@ -6,56 +6,81 @@ import Action from './action.jsx';
 import PanelActions from './panel_actions.jsx';
 
 import createStyler from '../util/create_styler';
+import formatRoute from '../util/format_route';
 
 import panelCss from '../css/panel.css';
 import routeFormCss from '../css/route_form.css';
 
 const propTypes = {
   domains: React.PropTypes.array.isRequired,
   route: React.PropTypes.object,
-  handleSubmit: React.PropTypes.func,
-  handleCancel: React.PropTypes.func
+  submitHandler: React.PropTypes.func,
+  cancelHandler: React.PropTypes.func,
+  deleteHandler: React.PropTypes.func
 };
 
 const defaultProps = {
   route: {},
-  handleSubmit: () => {},
-  handleCancel: () => {}
+  submitHandler: () => {},
+  cancelHandler: () => {}
```

Comment:
The form doesn't always have a delete handler. For example, when creating a route there is nothing to delete